### PR TITLE
Add support for alarm_control_panel entities

### PR DESCRIPTION
**Pull request by grahambrown11**
# Alarm Control Panel API

Implement API for new `alarm_control_panel` component

Related PRs:
esphome/esphome#4770
home-assistant/core#92357

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
     long_description = readme_file.read()
 
 
-VERSION = "13.7.4"
+VERSION = "13.8.0"
```
Comment (bdraco) [inline setup.py:13]: No need to change this. The tooling will

Comment (grahambrown11) [reply setup.py:13]: thanks @bdraco should I remove it?

File: aioesphomeapi/api.proto [hunk 2]
```diff
@@ -1461,3 +1462,62 @@ message VoiceAssistantEventResponse {
...
+  ALARM_CONTROL_PANEL_DISARM = 0;
+  ALARM_CONTROL_PANEL_ARM_AWAY = 1;
+  ALARM_CONTROL_PANEL_ARM_HOME = 2;
+  ALARM_CONTROL_PANEL_ARM_NIGHT = 3;
+  ALARM_CONTROL_PANEL_ARM_VACATION = 4;
+  ALARM_CONTROL_PANEL_ARM_CUSTOM_BYPASS = 5;
+}
+
+message ListEntitiesAlarmControlPanelResponse {
+  option (id) = 93;
```
Comment (jesserockz) [inline aioesphomeapi/api.proto:1490]: Please bump all the ids by +1 as I forgot to skip these ids with #439

Comment (grahambrown11) [reply aioesphomeapi/api.proto:1490]: sure updating now

**Review by jesserockz — APPROVED** [review]

**Pull request merged**